hotness/config.py

```diff
@@ -0,0 +1,172 @@
+        "bug_status": "NEW",
+        "explanation_url": "https://fedoraproject.org/wiki/upstream_release_monitoring",
+        "short_desc_template": "%(name)s-%(latest_upstream)s is available",
+        "description_template": """
+Latest upstream release: %(latest_upstream)s
+
+Current version/release in %(repo_name)s: %(repo_version)s-%(repo_release)s
+
+URL: %(url)s
+
+
+    Please consult the package updates policy before you
+issue an update to a stable branch:
+    https://fedoraproject.org/wiki/Updates_Policy
+
+
+More information about the service that created this bug can be found at:
+
+    %(explanation_url)s
+
+
+    Please keep in mind that with any upstream change, there may also be packaging
+    changes that need to be made. Specifically, please remember that it is your
+    responsibility to review the new version to ensure that the licensing is still
+    correct and that no non-free or legally problematic items have been added
+    upstream.
+
+Based on the information from anitya: https://release-monitoring.org/project/%(projectid)s/
+""",
+    },
+    # Koji configuration
+    KOJI={
+        "server": "https://koji.fedoraproject.org/kojihub",
+        "weburl": "https://koji.fedoraproject.org/koji",
+        # Kerberos configuration to authenticate with Koji. In development
+        # environments, use `kinit <fas-name>@FEDORAPROJECT.ORG` to get a
+        # Kerberos ticket and use the default settings below.
+        "krb_principal": "",
+        "krb_keytab": "",
+        "krb_ccache": "",
+        "krb_proxyuser": "",
+        "krb_sessionopts": {"timeout": 3600, "krb_rdns": False},
+        "git_url": "https://src.fedoraproject.org/cgit/rpms/{package}.git",
+        "user_email": [
+            "Upstream Monitor",
+            "<upstream-release-monitoring@fedoraproject.org>",
+        ],
+        "opts": {"scratch": True},
+        "priority": 30,
+        "target_tag": "rawhide",
+        # These are errors that we won't scream about.
+        "passable_errors": [
+            # This is the packager's problem, not ours.
+            "unclosed macro or bad line continuation"
+        ],
+    },
+    # Anitya configuration
+    ANITYA={"url": "https://release-monitoring.org"},
+    # Cache configuration
+    CACHE={
+        "backend": "dogpile.cache.dbm",
+        "expiration_time": 300,
+        "arguments": {"filename": "/var/tmp/the-new-hotness-cache.dbm"},  # nosec
+    },
+)
+
+
+def load(config_dict: dict) -> dict:
+    """
+    Load application configuration from a file and merge it with default configuration.
+
+    Params:
+        config_dict: Dictionary with configuration
+
+    Returns:
+        Merged configuration dictionary.
+    """
+    config = DEFAULTS.copy()
+
+    consumer_config = config_dict["consumer_config"]
+
+    _log.info("Loading the-new-hotness configuration")
+    for key in consumer_config:
+        try:
+            if config[key.upper()] is dict:
```

Comment:
Wait, this is so that we can write config keys case-insensitive?

Reply:
Yes, it will allow you to write config keys case-insensitive.